### DISCUSSION TITLE
"LIKE" Search in Cottontaildb

### DISCUSSION THREAD

**Discussion by amohap** [vitrivr/cottontaildb#78] [Q&A] [answered]
Dear devs,

We are currently implementing cottontaildb using Python. We ran into an issue with the "LIKE" search. 

So far, it was possible to execute the "EQUAL" search for literals, but the "LIKE" search only yields results if the input is given exactly the same way as for the "EQUAL" search. 

For example, the word _"nomodel"_ is in our database. With the "EQUAL" search it outputs "nomodel", but when trying to perform the "LIKE" search with input _"nomode"_ it gives no results. (see screenshot attached). 

Do we need to specify anything else for the like-search that is not implemented for the equal search?

Our source code can be found here (query on line 402): https://github.com/lutharsanen/tal-backend/blob/main/cottontaildb/cottontaildb_client/cottontaildb_client.py

![EQUAL_vs_LIKE_search](https://user-images.githubusercontent.com/71710151/118632371-a2f55600-b7d0-11eb-8aec-e2d202db241c.PNG)

Best,
Adriana

**Comment by ppanopticon** [comment, marked as answer]
Hey! Unfortunately there is no link to your source code. Hence, I can't find it.

**Reply by amohap** [reply]
Hey, sorry here is the link to the repo: https://github.com/lutharsanen/tal-backend/blob/main/cottontaildb/cottontaildb_client/cottontaildb_client.py

In line 402 is the start of the function. In line 423 we specify the atomic boolean predicate.

Best,
Adriana

**Reply by ppanopticon** [reply]
I can't really tell from the source code since this seems pretty boiler plate to me. However, LIKE uses the same Syntax as in SQL. That is, if you want to do a wildcard search then you must indicate this with a % character:

- LIKE 'nomodel' is equivalent to EQUAL 'nomodel'
- LIKE 'nomode' is equivalent to EQUAL 'nomode'
- LIKE 'nomode%' is equivalent to BEGINS WITH 'nomode' (obviously, BEGINS WITH is not a real operator)

Do you add the wildcard anywhere?

**Reply by amohap** [reply]
Hey,
Thanks for clarifying the syntax of the LIKE search. We did not include the wildcard '%' and therefore, it couldn't search for the correct term. This solved our issue.

Best,
Adriana
